lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -133,6 +133,7 @@ class RenderBlockingResources extends Audit {
     const traceOfTab = await TraceOfTab.request(trace, context);
     const simulator = await LoadSimulator.request(simulatorData, context);
     const wastedCssBytes = await RenderBlockingResources.computeWastedCSSBytes(artifacts, context);
+    TraceOfTab.assertIsNavigation(traceOfTab);
```

Comment:
part of why we made FCP required was because (at the time) _everything_ had to check if FCP was actually in the trace to make sure we weren't auditing nonsense. We've mitigated that in some other ways since, but I worry that we'd be regressing that by going back to assertions in every audit. In an ideal world everything works only if it's being done correctly. What if an audit forgets to call this but only implicitly relies on FCP being defined? Would we notice the somewhat nonsense numbers coming out? (this kind of gets back to my worry about test coverage * audits * mode).

You're in the thick of this so you'd know the feasibility better than me just handwaving here, but what if we changed a little bit more about the model here? e.g. rather than conditionals in trace-of-tab/trace-processor we split trace-processor so we have the thing everyone uses to identify the main process, frame tree, etc. then navigation auditing can pull in the timing/timestamp stuff from a separate module with that as input if/when needed? Almost all of that isn't needed for timespans, and when we get to the point where timespans can have an lcp as well, I'm ok with using a different model than coming in on a giant `timestamps` object :)

Reply:
> A total fork of trace-processor is certainly possible, but I'm a little fuzzy on the primary benefit. Is the goal for the type signature of trace output in timespan audits to not have a key at all for `firstContentfulPaint` instead of an optional one?

splitting it up, not forking, but yes :) timespan's portion of the trace-processor's output is a subset of what navigations need, so lets just only fetch timings/timestamps when they're needed, rather than always fetching them and asserting they're defined when needed.